app/src/main/java/com/codesoom/assignment/TaskHttpHandler.java

```diff
@@ -0,0 +1,188 @@
+package com.codesoom.assignment;
+
+import com.codesoom.assignment.models.Task;
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.sun.net.httpserver.HttpExchange;
+import com.sun.net.httpserver.HttpHandler;
+
+import java.io.*;
+import java.net.URI;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.stream.Collectors;
+
+public class TaskHttpHandler implements HttpHandler {
+    private List<Task> taskList = new ArrayList<>();
+    private ObjectMapper objectMapper = new ObjectMapper();
+    private Long curTaskID = 0L;
+    private final int SUCCESS = 200;
+    private final int CREATED = 201;
+    private final int DELETED = 204;
+    private final int NOT_FOUND = 404;
+
+    @Override
+    public void handle(HttpExchange exchange) throws IOException {
+        String method = exchange.getRequestMethod();
+        URI requestURI = exchange.getRequestURI();
+        String path = requestURI.getPath();
+
+        InputStream inputStream = exchange.getRequestBody();
+        String body = new BufferedReader(new InputStreamReader(inputStream))
+                .lines()
+                .collect(Collectors.joining("\n"));
+
+        if(method.equals("GET") && path.equals("/tasks")){
+            String content = taskListToJSON();
+            exchange.sendResponseHeaders(SUCCESS, content.getBytes().length);
+            sendResponseBody(exchange,content);
+        }
+
+        if(method.equals("GET") && path.startsWith("/tasks/")){
+            String searchIDString = path.substring("/tasks/".length());
+            System.out.println("ID String : " + searchIDString);
+            getRequest(exchange,searchIDString);
+        }
+
+        if(method.equals("POST") && path.equals("/tasks") && !body.isBlank()) {
+            postRequest(exchange, body);
+        }
+
+        if (method.equals("PUT") && path.startsWith("/tasks/") && !body.isBlank()) {
+            String searchIDString = path.substring("/tasks/".length());
+            System.out.println("ID String : " + searchIDString);
+            putRequest(exchange,searchIDString, body);
+        }
+
+        if (method.equals("DELETE") && path.startsWith("/tasks/")) {
+            String searchIDString = path.substring("/tasks/".length());
+            System.out.println("ID String : " + searchIDString);
+            deleteRequest(exchange, searchIDString);
+        }
+    }
+
+    private Long isValidURL(String IDString)  {
+        try {
+            Long searchID = Long.parseLong(IDString);
+            return searchID;
+        }
+        catch (Exception e){
+            throw e;
+        }
```

Comment:
```suggestion
        return Long.parseLong(IDString);
```
현재 코드와 제가 제안드린 코드와 어떤 점이 다를까요?!

Reply:
바로 return을 하면 숫자가 아닌 문제일 경우 자동적으로 exception이 발생하나요??

Reply:
변수를 생성하지 않고 바로 return 하는 차이점인거 같습니다

Reply:
맞아요! 차이점은 동일하죠! 그렇지만 Exception에 대한 회복이나 추가적인 행위를 하지않고 그대로 리턴하는 것이라면 
catch를 하는 코드와 안하는 코드는 동일합니다!